linode_metadata/metadata_client.py

```diff
@@ -70,6 +78,9 @@ def __init__(
         self.base_url = base_url
         self._append_user_agent = user_agent
         self.timeout = timeout
+        self._debug = debug
+        if debug:
+            logging.basicConfig(filename=debug_file, level=logging.DEBUG)
```

Comment:
Since this is using the global logger, it looks like I'm seeing some unrelated logs when debug mode is enabled. Could we scope the logger like the do [here](https://docs.python.org/3/howto/logging.html#configuring-logging)?

Reply:
Good catch! Let me look into this

Reply:
Added handler and formatter to better output the logs.